### ISSUE TITLE
BugsnagErrorBoundary type doesn't work with children with React 18 types

### ISSUE THREAD

**Issue by jarkko** [bugsnag/bugsnag-js#1717]
### Describe the bug

`React.FunctionalComponent` and equivalent types don't come with an implicit `children` prop anymore in React 18. This means that a `BugsnagErrorBoundary` component with children won't compile anymore. This could be fixed by explicitly adding `React.PropsWithChildren` to the type like this:

```typescript
export type BugsnagErrorBoundary = React.ComponentType<React.PropsWithChildren<{
  onError?: OnErrorCallback
  FallbackComponent?: React.ComponentType<{
    error: Error
    info: React.ErrorInfo
    clearError: () => void
  }>
}>>
```

### Steps to reproduce
1. In a Typescript React 18 app, create a component that wraps its contents inside an error boundary:
```typescript
  const ErrorBoundary = Bugsnag.getPlugin("react")?.createErrorBoundary(React)
  return ErrorBoundary ? <ErrorBoundary>{children}</ErrorBoundary> : null
```
2. try to compile the component.
3. See error:
```
      TS2559: Type '{ children: ReactNode; }' has no properties in common with type 'IntrinsicAttributes & { onError?: OnErrorCallback | undefined; FallbackComponent?: ComponentType<{ error: Error; info: ErrorInfo; clearError: () => void; }> | undefined; }'.
```

### Environment
* Bugsnag version:
  * "@bugsnag/browser": "^7.16.2",
  * "@bugsnag/plugin-react": "^7.16.2"
* Browser framework version (if any):
    * React: 18.0.1

**Comment by luke-belton** [comment]
Hi @jarkko - thanks for raising this, we're going to investigate and we'll keep you updated on this thread!

**Comment by jake-engel** [comment]
@luke-belton  What is the progress on this issue? I am running into it as well

**Comment by seanmcquaid** [comment]
Also seeing this with React 18!

**Comment by luke-belton** [comment]
Hi @jake-engel & @seanmcquaid - [a fix](https://github.com/bugsnag/bugsnag-js/pull/1723) for this was released in [`v7.16.4` ](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.16.4)of `bugsnag-js` 🎉
